resources/contribute-documentation/tutorial-guidelines.md

```diff
@@ -1,84 +1,129 @@
 
-Sample code is desirable, because advanced users can typically scan the example and use it immediately without a formal tutorial. It can also be used by others as a basis for tutorials. Sample code developers can focus on what they do well, while technical writers and support personnel can use the samples to create quality training materials.
+For example:
 
+```
+- _code_samples/issue-credentials/
+    - js/
+        - README.md
+        - issue-credential.js
+        - package.json
+    - py/
+        - README.md
+        - issue_credential.py
+        - requirements.txt
+    - README.md
+```
 
-## Tasks
+This information is used to populate the [Code Samples](/resources/code-samples/) page. The outer readme's header is the title of the code sample on that page, and the first paragraph is the description.
 
-Tasks are step-by-step instructions for how to accomplish a specific result. For example, “Installing rippled on a Red Hat Linux Server.” Task documentation is not intended to be particularly educational. It frequently describes tasks that are only performed one time per implementation, or maintenance tasks that always follow a familiar pattern. Tasks provide troubleshooting guidance, since there are likely variables that the user must adjust based on the specifics of their use case.
+### Comments
 
+Comments are an important part of readable, accurate code, but don't go overboard. The text of a tutorial may be translated into other languages such as Japanese, while the sample code is kept the same, so don't include any critical information _only_ in the comments.
 
-## Concepts
+Use comments to separate out logical sections of the sample code. You can these comments as markers so that the `{% code-snippet ... %}` Markdoc component shows only the relevant section at a time in a tutorial.
 
-Conceptual information describes elements of the API, how they work, and when to use them. If a tutorial requires lengthy explanations before or during the programming tasks, consider how you might separate the exposition into a new topic, or link to existing topics that set the proper context.
+### Sample Code Types
 
-For example, three paragraphs of context and a single line of code would be a concept, not a tutorial.
+Sample code can take many forms, such as the following:
 
+- **Sample Application** - A fully functional program that accepts user input to perform tasks with some flexibility. It may have a graphical user interface or only a commandline interface. Complete sample code for different stages of application development may or may not be provided.
+- **Script** - A simple program that performs a predetermined set of tasks, often using hard-coded values, with minimal branching. These scripts are not expected to perform robust error handling; they typically exit when an error occurs.
```

Comment:
Glad we've clarified this here, I had questions about the level of error handling we needed for scripts and this answers it pretty much 👍🏾 .